projects/observability/src/pages/custom-dashboards/custom-dashboards-view.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { NavigationParamsType, NavigationService } from '@hypertrace/common';
+import { ButtonRole, ButtonStyle, ToggleItem } from '@hypertrace/components';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./custom-dashboards.component.scss'],
+  template: `
+    <div class="custom-dashboards">
+      <div class="title">
+        <h2>Custom Dashboards</h2>
+        <ht-link [paramsOrUrl]="this.currentContext.value + '/create'" class="create-dashboard-button">
+          <ht-button role="${ButtonRole.Primary}" display="${ButtonStyle.Solid}" label="Create Dashboard"> </ht-button>
+        </ht-link>
+      </div>
+      <ht-toggle-group
+        class="context-data-toggle"
+        [items]="this.contextItems"
+        [activeItem]="this.currentContext"
+        (activeItemChange)="this.onContextChange($event)"
+      ></ht-toggle-group>
+      <router-outlet></router-outlet>
+    </div>
+  `
+})
+export class CustomDashboardsViewComponent {
+  public currentContext: ToggleItem = {};
+  public readonly contextItems: ToggleItem<string>[] = [
+    {
+      label: 'My Dashboards',
+      value: DASHBOARD_VIEWS.MY_DASHBOARDS
+    },
+    {
+      label: 'All Dashboards',
+      value: DASHBOARD_VIEWS.ALL_DASHBOARDS
+    }
+  ];
+  public constructor(private readonly navigationService: NavigationService) {
+    this.currentContext = this.contextItems[0];
```

Comment:
I think we should read from `url` here.
If I am at `https://hypertrace.concierge.stage.razorpay.in/custom-dashboards/all-dashboards?time=1h` and I reload the page, it always redirects or resets to `my-dashboards`.

What are your thoughts?